### PR TITLE
Battle environment overrides

### PR DESCRIPTION
**Pull request by Aegyo**
Battle environment overrides for places that don't want to just use the town environment. Blue 1 temp battle and Kanto Elite 4 used as examples.

- Adds environment optional arg to Gym and TemporaryBattle
- Moves gym displayName and imageName into new optionalArgs
- Gym rewardFunction and gymFlags should probably also be moved into optionalArgs, but I didn't want the hassle of making sure that all still works

### CODE REVIEW

File: src/scripts/towns/TownList.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 ///<reference path="../quests/BulletinBoard.ts"/>
 ///<reference path="BattleCafe.ts"/>
 ///<reference path="../../declarations/requirements/MultiRequirement.d.ts"/>
-///<reference path="../Safari/SafariTownContent.ts"/>
+///<reference path="../safari/SafariTownContent.ts"/>
```
Comment (osherwott) [inline src/scripts/towns/TownList.ts:8]: This should probably be its own PR...

**Comment by Jaaslet** [comment]
Why are the values in `optionalGymArgs` not just a part of `gymFlags`?

**Comment by Aegyo** [comment]
Because flag means a boolean value, which the things in optional args are not

**Review by Jaaslet — APPROVED** [review]
> Because flag means a boolean value, which the things in optional args are not

You are correct, and I don't like it...

Looks good. Thanks for the PR
